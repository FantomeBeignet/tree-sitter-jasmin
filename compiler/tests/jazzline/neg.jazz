param int Li2_q = 10;

fn _neg (reg u64 v64) -> reg u64
requires #[prover=smt] { v64 <= 100 }
ensures #[prover=cas] { result.0 == result.0}
{
reg u64 v64s;

#[tran = smt] v64s = v64 * (Li2_q);
v64s = -v64s;

return v64s;
}
