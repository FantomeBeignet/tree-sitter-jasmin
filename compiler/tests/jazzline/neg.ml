open Common

let () = parse_and_print "neg.jazz"  "_neg"
