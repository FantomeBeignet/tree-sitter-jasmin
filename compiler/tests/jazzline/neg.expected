proc _neg(v64s_140@uint64, v64_132@uint64) = 
{
 true && (v64_132@uint64 <= (const 64 (100)))
}
mull TMP_____151@uint64 TMP_____152@uint64 v64_132@uint64 (10)@uint64;
cast v64s_139@uint64 TMP_____152@uint64;
cast TMP_____153@sint64 v64s_139@uint64;
sub TMP_____154@sint64 (0)@sint64 TMP_____153@sint64;
cast v64s_140@uint64 TMP_____154@sint64;
{
 /\[(v64s_140 = v64s_140)] && true
}
 